src/components/Stake/Sections.tsx

```diff
@@ -0,0 +1,73 @@
+import { BigNumber } from "ethers/lib/ethers";
+import { FC } from "react";
+import * as StakeV2Styled from "src/views/Stake/StakeV2Styles";
+import { TokenIcon } from "src/components/Stake/TokenIcon";
+import { CompatibleToken, TokenSize, TokenSizeEnum } from "src/components/Stake/types";
+import { convertStringToFloat, parseBigNumberToString, formatNumberWithCommas } from "src/utils/common";
+import { expandDecimals } from "src/Helpers";
+
+interface WalletBalanceProps {
+  large?: boolean;
+  walletAmount: BigNumber;
+  tokenUsdPrice: BigNumber;
+  selectedToken: CompatibleToken;
+  tokenSize: TokenSize;
+  tokenDecimals?: number;
+  decimals?: number;
+}
+
+export const USD_PRICE_PRECISION = 30;
+
+export const WalletBalance: FC<WalletBalanceProps> = ({
+  large,
+  walletAmount,
+  tokenUsdPrice,
+  selectedToken,
+  tokenSize = TokenSizeEnum.lg,
+  decimals = 2,
+}) => (
+  <>
+    <StakeV2Styled.FlexColEnd>
+      <TokenAmountRow
+        large={large}
+        tokenAmount={parseBigNumberToString(walletAmount, decimals)}
+        selectedToken={selectedToken}
+        tokenSize={tokenSize}
+        decimals={decimals}
+      />
+      <StakeV2Styled.Subtitle>
+        $
+        {formatNumberWithCommas(
+          parseFloat(
+            parseBigNumberToString(tokenUsdPrice.mul(walletAmount).div(expandDecimals(1, USD_PRICE_PRECISION)), 2)
```

Comment:
Should probably just use formatAmount here

Reply:
Yeah got a bit fatigued at the end, that's a better option for sure